src/libs/actions/Session/index.js

```diff
@@ -310,6 +310,24 @@ function beginAppleSignIn(idToken) {
     API.write('SignInWithApple', {idToken}, {optimisticData, successData, failureData});
 }
 
+/**
+ * Given an idToken from Sign in with Apple, we call beginAppleSignIn while also setting the loading state
+ *
+ * @param {String} idToken
+ * @param {Function} setLoading
+ */
+
+function setLoadingAndSignInWithApple(token, setLoading) {
```

Comment:
Instead of adding a new function we can reuse the beginAppleSignIn like

`export const beginAppleSignIn = (token, callback) => {
    Session.beginAppleSignIn(token)
        .then(() => {
            if (typeof callback === 'function') {
                callback(); // callback will be setLoading(false)
            }
        })
        .catch((error) => {
            console.error(error);
            if (typeof callback === 'function') {
                callback();
            }
        });
};`

Reply:
Very true! Good thought @curlyCoder15